### PR TITLE
fix the bug of one hit per layer. cut

### PR DESCRIPTION
**Pull request by CollinKa**
I noticed the oneHitPerLayerCut() method has an issue. The count() in the awkward array can only count the size of arrays. 

Here is the quick test script to replicate the issue:
-----------------------------------
import awkward as ak

array = ak.Array([[1, 0, 3], [0, 2, 0], [0, 2, 0]]) # The inner arrays represent the pulse layer inside each event. It is the array from self.events.type.

print(ak.count(array == 0, axis=1)) #It return [3,3,3]

print(ak.count(array == 0, axis=1) == 1)#It return [False,False,Fales]. This replicates the issue from ak.count(self.events.layer==0, axis=1)==1

-----------------------------------
In short the previous "oneHitPerLayerCut"  checked if the event only contains one pulse instead of one hit per layer.
self.events['oneHitPerLayerCut'] =((ak.count(self.events.layer==0, axis=1)==1) &
(ak.count(self.events.layer==1, axis=1)==1) &
(ak.count(self.events.layer==2, axis=1)==1) &
(ak.count(self.events.layer==3, axis=1)==1))

To fix this bug I modified the oneHitPerLayerCut(), so it checks if the size of pulses for each event is equal to 4 and if the events pass the four layers cut.

Inside the demonstration file oneHITPL.py, the first cutflow that comes with "OLDoneHitPerLayerCut()" is to show bugs.

### CODE REVIEW

**Review by carriganm95 — CHANGES_REQUESTED** [review]
Hi Collin, you are right this is a bug but it can be fixed with the count_nonzero function instead of the changes you suggested. This will avoid loops

File: Run3Detector/analysis/utilities/milliqanCuts.py [hunk 1]
```diff
@@ -77,10 +78,16 @@ def fourLayerCut(self, cutName=None, cut=False):
 
     #event level mask selecting events with 1 hit in each layer
     def oneHitPerLayerCut(self, cutName=None, cut=False):
-        self.events['oneHitPerLayerCut'] =((ak.count(self.events.layer==0, axis=1)==1) &
```
[the remark this answers is not in the document]
Comment (CollinKa) [reply Run3Detector/analysis/utilities/milliqanCuts.py:80]: The issue is fixed in the latest commit.

File: Run3Detector/analysis/utilities/milliqanCuts.py [hunk 2]
```diff
@@ -75,12 +77,44 @@ def fourLayerCut(self, cutName=None, cut=False):
...
+
+    """
     #event level mask selecting events with 1 hit in each layer
     def oneHitPerLayerCut(self, cutName=None, cut=False):
-        self.events['oneHitPerLayerCut'] =((ak.count(self.events.layer==0, axis=1)==1) & 
-                                           (ak.count(self.events.layer==1, axis=1)==1) & 
-                                           (ak.count(self.events.layer==2, axis=1)==1) &
-                                           (ak.count(self.events.layer==3, axis=1)==1))
+        self.events['oneHitPerLayerCut'] =((ak.count_nonzero(self.events.layer==0, axis=1)==1) & 
+                                           (ak.count_nonzero(self.events.layer==1, axis=1)==1) &
```
Comment (carriganm95) [inline Run3Detector/analysis/utilities/milliqanCuts.py:115]: Just remove this big commented section now

**Review by carriganm95 — APPROVED** [review]

**Review by carriganm95 — APPROVED** [review]
